src/lang/typechecking.ml

```diff
@@ -220,14 +223,30 @@ let rec check ?(print_toplevel = false) ~throw ~level ~(env : Typing.env) e =
                        Meth
                          ( {
                              meth = l;
+                             optional = default <> None;
                              scheme = ([], x);
                              doc = "";
                              json_name = None;
                            },
                            y ));
-                x
+                ([], x)
         in
-        e.t >: aux a.t
+        let vars, typ = aux a.t in
+        let typ =
+          match default with
+            | None -> typ
+            | Some v ->
+                check ~level ~env v;
+                (* We want to make sure that: x?.foo types as: { foo?: 'a } *)
+                let typ =
+                  match (Type.deref v.t).descr with
+                    | Type.Nullable _ -> mk Type.(Nullable typ)
```

Comment:
If we have a universal variable here, isn't there a chance that we build a "double nullable" here? (like `int??`) I don't think we want that... For instance, we would have `'a?` and then `'a` gets substituted by `int?`. Or maybe should this be solved by a "not nullable" constraint on universal variables? I don't think we have to solve this *right now* but it would be nice to have a comment or keep a BR open so that we don't entirely forget about it.

Reply:
You might still want the double `null` I think, for instance if the default value is expected to be a nullable thing:
```ruby
# def f(x) =
    v = x?.foo
    if (null.defined(v)) then
      null.get(v) ?? 1
    else
      2
    end
  end;;
f : ('a.{foo? : int?}) -> int = <fun>
```

Reply:
Ha yeah no that doesn't work.. 😅